HOMEWORK 2

LUCA NICOLOSI 1000047978

In [2]:
# Importing librares 

import pandas as pd

import matplotlib.pyplot as plt

import numpy as np

import yfinance as yf 


In [15]:
# STEP 0

df_full = pd.read_csv('Symbol_Info_extended_2.csv', sep=';', encoding='latin1') # I'm importing the dataset. With sep I'm specifing that each column is separated by a semicolon, while econding allow me to avoind problems with accents.

countries = ['Ireland', 'Italy', 'Germany', 'Netherlands', 'Luxembourg', 'Spain', 'France'] # I'm creating a list which includes all the countries which belong to group 5

df_filtered_group5 = df_full[df_full['country'].isin(countries)].copy()# I'm filtering the dataset by considering only the companies located in the selected countries

selected_symbol = 'STM'

company_info = df_filtered_group5[df_filtered_group5['symbol'] == selected_symbol]

print(company_info)


     symbol             company_name      sector        industry      country  \
2595    STM  STMicroelectronics N.V.  Technology  Semiconductors  Netherlands   

        market_cap   net_income  total_revenue  return_on_assets  \
2595  5.942478e+10  147000000.0   1.237800e+10           0.01523   

      return_on_equity  ...  pe_trailing  pe_forward  earnings_growth  \
2595           0.00908  ...      417.875   29.982063           -333.0   

      price_to_sales  price_to_book  revenue_growth  debt_to_equity  \
2595        4.800838       3.344505            0.23          15.316   

      dividend_yield  payout_ratio  free_cashflow  
2595            0.54          2.25   -360500000.0  

[1 rows x 21 columns]


The reasons why I selected STMicroelectronics are based on its market capitalization of 59.42 billion, which provides a high-scale anchor for European peer analysis. 

Despite a cyclical downturn compressing its Return on Assets to 1.52% and Return on Equity to 0.91%, its revenue growth remains resilient at +23%. This operational disconnect explains the highly distorted trailing P/E of 417.88x compared to a forward P/E of 29.98x, signaling market expectations of a cyclical recovery. 

Furthermore, its conservative Debt-to-Equity ratio of 15.32% allows the company to fund heavy capital investments, directly explaining the temporary negative Free Cash Flow of -$360.5 million.

In [ ]:
# STEP 1

required_var = ['symbol', 'company_name', 'sector', 'industry', 'country', 'market_cap',
    'return_on_assets', 'return_on_equity', 'profit_margins',
    'pe_trailing', 'price_to_book', 'revenue_growth', 'debt_to_equity', 'free_cashflow'] # I'm creating a list containing the name of all the requierd variables

df_clean = df_full[required_var].copy() # I'm creating a new dataset which includes just the required variables 

df_clean.replace([np.inf, -np.inf], np.nan, inplace=True) # I'm substituting infinite values with nan values

df_clean = df_clean[df_clean['market_cap'] > 0] # I'm filtering the dataset by excluding the companies with a non-positive market capitalization

df_clean.loc[df_clean['pe_trailing']<= 0, 'pe_trailing'] = np.nan # I'm substituting the negative values of the pe_trailing variable with nan values 
df_clean.loc[df_clean['price_to_book'] <= 0, 'price_to_book'] = np.nan # I'm substituting the negative values of the price_to_book variable with nan values

df_clean[df_clean['symbol'] == 'STM']


,symbol,company_name,sector,industry,country,market_cap,return_on_assets,return_on_equity,profit_margins,pe_trailing,price_to_book,revenue_growth,debt_to_equity,free_cashflow
2595,STM,STMicroelectronics N.V.,Technology,Semiconductors,Netherlands,5.942478e+10,0.01523,0.00908,0.01188,417.875,3.344505,0.23,15.316,-360500000.0
